services/py-api/src/server/handlers/participants_handlers.py

```diff
@@ -21,14 +25,27 @@ def __init__(self, service: ParticipantRegistrationService) -> None:
         self._service = service
 
     async def create_participant(
-        self, response: Response, input_data: ParticipantRequestBody
+        self, response: Response, input_data: ParticipantRequestBody, jwt_token: Optional[str] = None,
     ) -> ParticipantRegisteredResponse | ErrResponse:
+        
         if input_data.is_admin and input_data.team_name:
             result = await self._service.register_admin_participant(input_data)
 
-        # TODO: Implement this when the invite participant case is done
-        # elif input_data.is_admin is False and input_data.team_name:
-        #     result = await self._service.register_invite_participant(input_data)
+        elif input_data.is_admin is False and input_data.team_name:
+            # Decode the token
+            decoded_result = JwtUtility.decode_data(token=jwt_token, schema=JwtUserData)
+            if is_err(decoded_result):
+                response.status_code = status.HTTP_401_UNAUTHORIZED
+                return ErrResponse(error=f"JWT decoding failed: {decoded_result.err_value}")
+            
+            decoded_data = decoded_result.ok_value
+            team_name_from_token = decoded_data["team_id"]
+
+            if input_data.team_name != team_name_from_token:
+                response.status_code = status.HTTP_400_BAD_REQUEST
+                return ErrResponse(error="Team name in request does not match the invite link.")
```

Comment:
Move the token validation logic inside of the service layer. If you have any issues with decoding the token catch the exceptions in the handler. Remember we use the services to write all the business logic.

Only 
```Python
result = await self._service.register_invite_link_participant(input_data)
``` 

Should stay in the handler

Reply:
Fixed in commit a9a6b60a1c533cba63bd89fc5bee4362218c00f7